File: Problems/No.206_Reverse_Linked_List/reverse_linked_list_iterative.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        
        // parameter
        ListNode *reversed_head, *next_head;
        
        // start computing
        reversed_head = NULL;
        while (head != NULL) {

            next_head = head->next;
            head->next = reversed_head;
            reversed_head = head;
            head = next_head;

        }

        return reversed_head;

    }

File: Problems/No.206_Reverse_Linked_List/reverse_linked_list_iterative.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """

        # parameters
        reversed_head, next_head = None, None

        # start computing
        while head:
            next_head = head.next
            head.next = reversed_head
            reversed_head = head
            head = next_head
        
        return reversed_head
